src/sentry/runner/initializer.py

```diff
@@ -215,10 +221,8 @@ def apply_legacy_settings(settings):
 
     # Set ALLOWED_HOSTS if it's not already available
     if not settings.ALLOWED_HOSTS:
-        from urlparse import urlparse
-        urlbits = urlparse(settings.SENTRY_URL_PREFIX)
-        if urlbits.hostname:
-            settings.ALLOWED_HOSTS = (urlbits.hostname,)
+        from .hacks import AllowedHosts
+        settings.ALLOWED_HOSTS = AllowedHosts()
```

Comment:
future followup i think we should kill Django's internal protection and implement our own (or listen to changes on system.url-prefix and define ALLOWED_HOSTS from it, though people might need to override that value)